src/DriverManager.php

```diff
@@ -451,11 +454,26 @@ private static function parseDatabaseUrlScheme(?string $scheme, array $params):
             // URL schemes must not contain underscores, but dashes are ok
             $driver = str_replace('-', '_', $scheme);
 
-            // The requested driver from the URL scheme takes precedence over the
-            // default driver from the connection parameters. If the driver is
-            // an alias (e.g. "postgres"), map it to the actual name ("pdo-pgsql").
+            // If the driver is an alias (e.g. "postgres"), map it to the actual name ("pdo-pgsql").
             // Otherwise, let checkParams decide later if the driver exists.
-            $params['driver'] = self::$driverSchemeAliases[$driver] ?? $driver;
+            if (isset(self::$driverSchemeAliases[$driver])) {
+                $actualDriver = self::$driverSchemeAliases[$driver];
+
+                Deprecation::trigger(
+                    'doctrine/dbal',
+                    'https://github.com/doctrine/dbal/pull/TODO',
```

Comment:
Woopsie

Reply:
Good, it's not only me who forgets these things. 😅

Reply:
We've all done it I'm afraid :slightly_smiling_face: